src/game.cpp

```diff
@@ -5068,7 +5068,7 @@ void Game::playerCreateMarketOffer(uint32_t playerId, uint8_t type, uint16_t spr
 	}
 
 	if (g_config.getBoolean(ConfigManager::MARKET_PREMIUM) && !player->isPremium()) {
-		player->sendMarketLeave();
+		player->sendStoreError(STORE_ERROR_INFORMATION, "Only premium accounts may create offers for that object.");
```

Comment:
```suggestion
		player->sendMarketLeave();
		player->sendTextMessage(MESSAGE_MARKET, "Only premium accounts may create offers for that object.");
```

Reply:
This behavior is wrong. The correct thing is to remove sendMarketLeave. Keeping it will close the market after saying "ok" in the message window. What doesn't make sense, after all, it's just a message, why should I close the market? It would be the equivalent of closing the trade after purchasing an item.

Reply:
![image](https://user-images.githubusercontent.com/8551443/147707123-b70e5be0-e22c-4b0f-8117-ee36f516475a.png)
After press ok:
![image](https://user-images.githubusercontent.com/8551443/147707128-cfbc4048-888e-4cd0-a45b-a6c4354a10fc.png)